### ISSUE TITLE
Adding Binance

### ISSUE THREAD

**Issue by theimo1221** [pmaji/eth_python_tracker#29]
An issue to collect data and tasks needed for adding binance

- [ ]  Analyze compatibility
Needed changes in Data call?
Needed changes in Data storage (Multiple times same pair e.g. "ETH-USD")

- [ ]  Discuss form of presentation
Send all data to client, and hide/ show selection on clientside with js?

- [ ]  Add Dropdown to select exchange
When we change this we should change #28 aswell

Thanks and Greets

Theimo

**Comment by pmaji** [comment]
Yep, good idea and a big step forward. I read the Binance API info last night and it looks like a VERY different process that require a lot more work. 

I think it best to put this on the back-burner in terms of development priorities and ensure we lock down the methodology for GDAX first, but anyone is welcome to give other-exchange integration a shot.

**Comment by theimo1221** [comment]
If it´s so different, we should maybe consinder changing the pairs to Objejtc

- [ ] Pair to Object 
- [ ] Passing Object to get_data to get individual hooks from apis

Additionally we should consider getting the update into 1 Package, except of having 1 for each pair,  to highly decrease server load for Web Hosting (Further improvement of #10)

**Comment by MarkyVee** [comment]
Hey guys...doing great work. Any way of "time stamping" the transactions?

**Comment by pmaji** [comment]
@MarkyVee definitely something that has been considered and is not permanently something I don't PLAN to do, but I have tested the timestamp validity of the websocket multiple times and it is outside of the realm of what I would consider useful in terms of its trustworthiness. I want to be sure that I can stand behind whatever we report, and I don't have that degree of comfort yet with the timestamps. This is well documented issue on Stack and also in discussions with the folks at GDAX, so it may change in the future, but isn't an update I'm planning short-term.

**Comment by MarkyVee** [comment]
Thank you all for your responses. I’ll wait for the next update to mention the tool then. 
I’ll keep in touch through telegram. 

MarkyVee

**Comment by pmaji** [comment]
@MarkyVee It may actually may sense not to wait; the present state of hosting is perfectly stable, it's just a question of adding further additions and increased speed. Let's chat more via telegram so I can explain, but for your purposes I don't believe it makes sense to wait to mention it.

**Comment by pmaji** [comment]
Will keep this open in case a developer comes along who would like to tackle the challenge of integrating with Binance / other exchanges beyond coinbase, but I don't presently have time to tackle this. I also think that such an endeavor would be best accomplished via a separate fork of this code as the app is pretty computationally heavy as is.

**Comment by 3raserHead** [comment]
Hi, pmagi, Gdax is not really used anymore by traders, the majority moved to Binance and Bybit... It would be awesome to have the possibility to have this amazing tool on Binance and have the possibility to monitor each pairs on that exchange... 

I hope one this will be implemented in future!

Thank you all for developing this precious tool!

**Comment by pmaji** [comment]
Thanks for this @3raserHead. I agree that that would be supremely helpful, but sadly, it requires a decent amount of time which I don't currently have to refactor the repo to make it extensible to all other exchanges. I keep this issue open hoping for other volunteer contributors to take a wheel, but sadly don't have time to invest in this myself right now.

**Comment by 3raserHead** [comment]
Hi Paul,

First thank you very much for your precious work, now I understand better how the market is manipulated.. Are you working on similar tools at the moment? Nowadays tracking whales movements and tracking liquidation is key...

I’ll try to make a virtual environment, I’m new to python, I still need to get my head around it….

Thank you for everything,

Have a great evening,

Best,

Fede

f.nitti@me.com
www.federiconitti.net

**Comment by jihadd1987** [comment]
hi - thank you all  - i want to ask how to get this tool ? whale crypto ? the website not working ( https://whales.cracklord.com)
thank  you - i was using this tool before some years - now the website not working

**Comment by jihadd1987** [comment]
hi
